packages/uicore/src/components/RadioButton/RadioButton.tsx

```diff
@@ -30,11 +31,19 @@ export function RadioButton({
   checked = false,
   disabled = false,
   tooltipId,
+  asPill = false,
   onChange = () => undefined,
   ...props
 }: RadioButtonProps & StyledProps): ReactElement {
   return (
-    <label className={styledClasses(props, css.radio, className)} {...omitStyledProps(props)}>
+    <label
+      className={styledClasses(
+        props,
+        asPill ? css.asPill : css.radio,
+        asPill && checked ? css.pillChecked : '',
```

Comment:
This looks a bit cumbersome to me - it'd be neat if we could extract out a `type` variable to remove the inline `asPill ? css.asPill : css.radio` ternary, and make use of the input's `:checked` state to add the `pillChecked` styles, but I'm not sure how difficult that is to do - might be handiest to leave it all as-is!

Reply:
When `has` is available we can move all this to CSS using `.asPill:has(input:checked)`, which would be awesome!